packages/dd-trace/src/encode/0.4.js

```diff
@@ -263,6 +271,45 @@ class AgentEncoder {
     }
   }
 
+  _encodeObject (bytes, value, circularReferencesDetector = new Set()) {
+    circularReferencesDetector.add(value)
+    if (Array.isArray(value)) {
+      return this._encodeObjectAsArray(bytes, value, circularReferencesDetector)
+    } else if (typeof value === 'object') {
+      return this._encodeObjectAsMap(bytes, value, circularReferencesDetector)
+    } else if (typeof value === 'string' || typeof value === 'number') {
+      this._encodeValue(bytes, value)
+    }
+  }
+
+  _encodeObjectAsMap (bytes, value, circularReferencesDetector) {
+    const keys = Object.keys(value)
+    const validKeys = keys.filter(
```

Comment:
3 cases (may not be relevant):
- `undefined` keys (`typeof 'undefined'`) won't be encoded
- `null` keys (typeof 'object') - first one will be included and the rest filtered, no? because first `null` will be included in `circularReferencesDetector`
- Date keys (typeof 'object') but `Object.keys(date)` is an empty array

Reply:
> undefined keys (typeof 'undefined') won't be encoded
This is expected. (I'm going to add also a test for `undefined`)

> null keys (typeof 'object') - first one will be included and the rest filtered, no? because first null will be included in circularReferencesDetector

Good catch, I am going to add a test also for this case.
```
> Object.keys(null)
Uncaught TypeError: Cannot convert undefined or null to object
    at Function.keys (<anonymous>)
```

> Date keys (typeof 'object') but Object.keys(date) is an empty array so it will end up with an empty value
I think that we can ignore this, Date objects are not expected here, if they are encoded as `{}` not a problem.

Reply:
ferpect